### PR TITLE
[fixes 40] Added ability to manage delegated roles.

### PR DESCRIPTION
**Pull request by scr-oath**

### CODE REVIEW

File: athenz/data_source_role.go [hunk 1]
```diff
@@ -29,6 +29,12 @@ func DataSourceRole() *schema.Resource {
 				Elem:     &schema.Schema{Type: schema.TypeString},
 				Set:      schema.HashString,
 			},
+			"trust": {
+				Type:        schema.TypeString,
+				Description: "The domain, which this role is trusted to",
```
Comment (dvirguttman) [inline athenz/data_source_role.go:34]: It should be fully qualified role name (<domain>.<role>)

Comment (scr-oath) [reply athenz/data_source_role.go:34]: What should be? I don't understand this comment - delegated roles trust another domain; not another role - the mechanism on the client domain end is for _their_ role to "assume_role" the delegated role in this domain.

Comment (dvirguttman) [reply athenz/data_source_role.go:34]: Indeed.
I've confused.
`(oath_tools) dguttman@F265P9343M calypso-ui (sd_aws) $ zms-cli help add-delegated-role
 syntax:
   -d domain add-delegated-role role trusted_domain
 parameters:
   domain       : name of the domain that role belongs to
   role         : name of the cross-domain/trust delegated role
   trust_domain : name of the cross/trusted domain name
 examples:
   -d coretech add-delegated-role tenant.sports.readers sports`

**Review by dvirguttman — APPROVED** [review]

**Comment by scr-oath** [comment]
[fixes #40]

**Comment by samuel-phan** [comment]
Could you also add the documentation for it please?
https://registry.terraform.io/providers/AthenZ/athenz/latest/docs/resources/athenz_role

**Pull request merged**